vcs_source.go

```diff
@@ -102,44 +89,17 @@ func (s *gitSource) exportVersionTo(v Version, to string) error {
 	return err
 }
 
-func (s *gitSource) listVersions() (vlist []Version, err error) {
-	s.baseVCSSource.lvmut.Lock()
-	defer s.baseVCSSource.lvmut.Unlock()
-
-	if s.cvsync {
-		vlist = make([]Version, len(s.dc.vMap))
-		k := 0
-		for v, r := range s.dc.vMap {
-			vlist[k] = v.Is(r)
-			k++
-		}
-
-		return
-	}
-
-	vlist, err = s.doListVersions()
+func (s *gitSource) listVersions() ([]PairedVersion, error) {
+	vlist, err := s.doListVersions()
 	if err != nil {
 		return nil, err
 	}
-
-	// Process the version data into the cache
-	//
-	// reset the rmap and vmap, as they'll be fully repopulated by this
-	s.dc.vMap = make(map[UnpairedVersion]Revision)
-	s.dc.rMap = make(map[Revision][]UnpairedVersion)
-
-	for _, v := range vlist {
-		pv := v.(PairedVersion)
-		u, r := pv.Unpair(), pv.Underlying()
-		s.dc.vMap[u] = r
-		s.dc.rMap[r] = append(s.dc.rMap[r], u)
-	}
-	// Mark the cache as being in sync with upstream's version list
-	s.cvsync = true
-	return
+	// Process version data into the cache and
+	s.dc.storeVersionMap(vlist, true)
+	return vlist, nil
 }
 
-func (s *gitSource) doListVersions() (vlist []Version, err error) {
+func (s *gitSource) doListVersions() (vlist []PairedVersion, err error) {
```

Comment:
Probably could, yeah. It's still used in `gopkginSource.listVersions()` (which reuses `gitSource`'s `doListVersions()`, then applies the filtering logic in the body of its `listVersions()` implementation). But the original method could be called by addressing the embedded field, so that's not really a reason to keep it...